### PR TITLE
Post-process STRINGPART

### LINKED ISSUE #15514: String interpolation followed by a dot is missing string literals
## Compiler version

`3.2.0-RC1`

## Minimized code

```Scala
object Main {
 s"Hello $Main.toStr!"
}
```

## Output

This is the tree representing the string interpolation:
```scala
 Select(
    qualifier = Apply(
      fun = Select(
        qualifier = Apply(
          fun = Select(
            qualifier = Select(
              qualifier = Select(qualifier = Ident(name = _root_), name = scala),
              name = StringContext
            ),
            name = apply
          ),
          args = List(
            Typed(
              expr = SeqLiteral(
                elems = List(Literal(const = ( = "Hello "))),
                elemtpt = ...
              ),
              tpt = ...
            )
          )
        ),
        name = s
      ),
      args = List(
        ...
      )
    ),
    name = <error>
  ),
```

Notice it only contains `"Hello "` literal.

## Expectation

```scala
 Select(
    qualifier = Apply(
      fun = Select(
        qualifier = Apply(
          fun = Select(
            qualifier = Select(
              qualifier = Select(qualifier = Ident(name = _root_), name = scala),
              name = StringContext
            ),
            name = apply
          ),
          args = List(
            Typed(
              expr = SeqLiteral(
                 elems = List(Literal(const = ( = "Hello ")), Literal(const = ( = ".toStr!")))
                elemtpt = ...
              ),
              tpt = ...
            )
          )
        ),
        name = s
      ),
      args = List(
        ...
      )
    ),
    name = <error>
  ),
```

The main part is ` elems = List(Literal(const = ( = "Hello ")), Literal(const = ( = ".toStr!")))`

### PR DESCRIPTION
**Pull request by som-snytt**
When getting the next part of an interpolation,
consisting of the literal part followed by the interpolated
identifier or expression, defer fetching the identifier
to postProcessToken. This ensures that lookahead at an
interpolation looks ahead only one token.

Fixes #15514

### CODE REVIEW

**Comment by som-snytt** [comment]
Added the assert on reset from the sibling PR.

File: compiler/src/dotty/tools/dotc/parsing/Scanners.scala [hunk 1]
```diff
@@ -751,6 +751,8 @@ object Scanners {
           closeIndented()
         case EOF =>
           if !source.maybeIncomplete then closeIndented()
+        case STRINGPART =>
+          finishStringPart()
```
Comment (som-snytt) [inline compiler/src/dotty/tools/dotc/parsing/Scanners.scala:755]: A Finnish string part would be the violins in Sibelius's Finlandia.

**Review by odersky — CHANGES_REQUESTED** [review]
I must admit I found this hard to follow, and a bit unsystematic, since postProcess usually does not fiddle with token details like that.

I believe it's best to simply forbid looking ahead beyond an INTERPOLATIONID. It turned out forbidding this revealed a bug in Parser where the real problem lied.
